File: Java/CajaSupermercado/src/Carrito.java
import java.util.ArrayList;
import java.util.List;

public class Carrito {
    private List<Producto> listaProductos;
    private int totalAcumulado;

    public Carrito(){
        this.listaProductos = new ArrayList<>();
        this.totalAcumulado = 0;

    
    }
    public void agregadorProductos(Producto p){
        listaProductos.add(p);
        totalAcumulado += p.getPrecio();
    }

    public List<Producto> getLista(){
        return listaProductos;
    }

    public int getTotal(){
        return totalAcumulado;
    }
}

File: Cpp/Tarea1/Carrito.cpp
#ifndef CARRITO_H
#define CARRITO_H
#include <vector>
#include "Producto.h"

class Carrito {
private:
    std::vector<Producto*> listaproductos;
    int totalAcumulado = 0;
public:
    void agregarProducto(Producto* p) {
        if(p) {
            listaproductos.push_back(p);
            totalAcumulado += p->getPrecio();
        }
    }
    // Retornamos por referencia constante para ser más eficientes
    const std::vector<Producto*>& getLista() const { return listaproductos; }
    int getTotal() const { return totalAcumulado; }
};
#endif